google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -623,6 +632,11 @@ ExecuteSqlRequest.Builder getExecuteSqlRequestBuilder(
     if (options.hasDataBoostEnabled()) {
       builder.setDataBoostEnabled(options.dataBoostEnabled());
     }
+    if (options.hasDirectedReadOptions()) {
+      builder.setDirectedReadOptions(options.directedReadOptions());
+    } else if (defaultDirectedReadOptions != null) {
```

Comment:
I think we can replace this `else if` condition with `else`? I think by default builder has this field set to `null`.

Reply:
I wouldn't prefer to assume what the builder sets the default to. By this way we are being explicit that we honor default values if and only if the request values are not sent and the defaults are non-null.